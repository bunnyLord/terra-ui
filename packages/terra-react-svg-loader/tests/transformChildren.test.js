import fs from 'fs';
import transformChildren from '../src/transformChildren';

describe('htmlToReactAttributes', () => {
  it('should return the svg children as string with jsx attributes', () => {
    const alertSvg = fs.readFileSync('tests/icons/alert.svg', 'utf-8');
    document.body.innerHTML = alertSvg;
    expect(transformChildren(document.body.children[0])).toMatchSnapshot();
  });

  it('should return the svg children as string with jsx attributes', () => {
    const guitarSvg = fs.readFileSync('tests/icons/guitar.svg', 'utf-8');
    document.body.innerHTML = guitarSvg;
    expect(transformChildren(document.body.children[0])).toMatchSnapshot();
  });

  it('should return the svg children as string with jsx attributes', () => {
    const redditSvg = fs.readFileSync('tests/icons/reddit.svg', 'utf-8');
    document.body.innerHTML = redditSvg;
    expect(transformChildren(document.body.children[0])).toMatchSnapshot();
  });
});

