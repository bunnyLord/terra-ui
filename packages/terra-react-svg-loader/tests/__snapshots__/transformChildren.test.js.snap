exports[`htmlToReactAttributes should return the svg children as string with jsx attributes 1`] = `"<path fill=\"#E50000\" d=\"M1.2 45c-1.1 0-1.6-.8-1-1.7L23 3.7c.5-1 1.4-1 2 0l22.8 39.6c.5 1 .1 1.7-1 1.7H1.2z\" ></path><path fill=\"#FFF\" d=\"M21.5 36.7h5V42h-5v-5.3zm0-22.7h5v17.3h-5V14z\" ></path>"`;

exports[`htmlToReactAttributes should return the svg children as string with jsx attributes 2`] = `"<g ><rect x=\"257.4\" y=\"472.1\" fill=\"#010101\" width=\"84.4\" height=\"27.9\" ></rect><path fill=\"#010101\" d=\"M374.9,96.2c25.8-3,23.1-31,19.5-46.9c-0.6-3-1.4-5.9-2.4-8.7c-0.4-1.2-0.6-2-0.6-2h-0.1   C382.5,16.1,360.6,0,334.8,0c-27.8,0-51.3,18.8-58.4,44.5h-0.2l-87.5,364.9c48.8,11.9,68.6,51.7,68.6,51.7h84.7v-15.5   c2.1-15.7,55.4-54.9,55.4-54.9C367.5,321.1,374.9,96.2,374.9,96.2z M260,373c-9.2,0-16.5-7.4-16.5-16.5s7.4-16.5,16.5-16.5   c9.2,0,16.5,7.4,16.5,16.5S269.1,373,260,373z M285.2,276c-9.2,0-16.5-7.4-16.5-16.5c0-9.2,7.4-16.5,16.5-16.5   c9.2,0,16.5,7.4,16.5,16.5C301.7,268.7,294.2,276,285.2,276z M301.9,179.2c-9.2,0-16.5-7.4-16.5-16.5s7.4-16.5,16.5-16.5   c9.2,0,16.5,7.4,16.5,16.5S311,179.2,301.9,179.2z M318.6,82.4c-9.2,0-16.5-7.4-16.5-16.5s7.4-16.5,16.5-16.5   c9.2,0,16.5,7.4,16.5,16.5S327.8,82.4,318.6,82.4z\" ></path><path fill=\"#010101\" d=\"M206.6,93.2c0.1,0,0.1,0,0.2,0c1.2,10.8,10.2,19.3,21.3,19.3c11.9,0,21.5-9.6,21.5-21.5   c0-0.2-0.1-0.5-0.1-0.6h0.1V78.1h10.7l5.8-24.4h-16.5V41.4h-0.1c0-0.2,0.1-0.5,0.1-0.6c0-11.9-9.6-21.5-21.5-21.5   c-11.1,0-20.1,8.4-21.3,19.3c-0.1,0-0.1,0-0.2,0c-15.1,0-27.4,12.3-27.4,27.4C179.1,81,191.4,93.2,206.6,93.2z\" ></path><path fill=\"#010101\" d=\"M181.6,190.1c0.1,0,0.1,0,0.2,0c1.2,10.8,10.2,19.3,21.3,19.3c11.9,0,21.5-9.6,21.5-21.5   c0-0.2-0.1-0.5-0.1-0.6h0.1V175h12.3l5.8-24.4h-18.1v-12.4h-0.1c0-0.2,0.1-0.5,0.1-0.6c0-11.9-9.6-21.5-21.5-21.5   c-11.1,0-20.1,8.4-21.3,19.3c-0.1,0-0.1,0-0.2,0c-15.1,0-27.4,12.3-27.4,27.4C154.3,177.8,166.5,190.1,181.6,190.1z\" ></path><path fill=\"#010101\" d=\"M158.1,287c0.1,0,0.1,0,0.2,0c1.2,10.8,10.2,19.3,21.3,19.3c11.9,0,21.5-9.6,21.5-21.5   c0-0.2-0.1-0.5-0.1-0.6h0.1v-12.3h12.4l5.8-24.4h-18.1v-12.3h-0.1c0-0.2,0.1-0.5,0.1-0.6c0-11.9-9.6-21.5-21.5-21.5   c-11.1,0-20.1,8.4-21.3,19.3c-0.1,0-0.1,0-0.2,0c-15.1,0-27.4,12.3-27.4,27.4C130.7,274.7,143,287,158.1,287z\" ></path><path fill=\"#010101\" d=\"M196,344.2h-23v-12.3h-0.1c0-0.2,0.1-0.5,0.1-0.6c0-11.9-9.6-21.5-21.5-21.5c-11.1,0-20.1,8.4-21.3,19.3   c-0.1,0-0.1,0-0.2,0c-15.1,0-27.4,12.3-27.4,27.4c0,15.1,12.3,27.4,27.4,27.4c0.1,0,0.1,0,0.2,0c1.2,10.8,10.2,19.3,21.3,19.3   c11.9,0,21.5-9.6,21.5-21.5c0-0.2,0-0.5-0.1-0.6h0.1v-12.3h17.1L196,344.2z\" ></path></g>"`;

exports[`htmlToReactAttributes should return the svg children as string with jsx attributes 3`] = `"<lineargradient gradientUnits=\"userSpaceOnUse\" x1=\"341.762\" y1=\"90.955\" x2=\"341.762\" y2=\"92.331\" gradientTransform=\"translate(-11666.8 -2982.078) scale(34.7225)\" ><stop offset=\"0\" stopColor=\"#FFF\" ></stop><stop offset=\"1\" stopColor=\"#EAEAEA\" ></stop></lineargradient><path fill=\"url(#a)\" d=\"M223.904 217.93c0 3.3-2.677 5.975-5.976 5.975h-35.856c-3.3 0-5.977-2.677-5.977-5.976v-35.857c0-3.3 2.676-5.977 5.977-5.977h35.856c3.3 0 5.976 2.677 5.976 5.977v35.856z\" ></path><g fillRule=\"evenodd\" clipRule=\"evenodd\" fill=\"#E6E6E6\" ><path d=\"M189.897 194.724c2.914-1.658 6.07-2.337 9.31-2.542.12-.954.15-1.837.35-2.68.79-3.31 3.165-4.682 6.668-3.772 1.127.292 1.867.438 2.934-.515 2.31-2.062 5.874-.683 6.583 2.326.498 2.12-1.342 4.51-3.56 4.617-2.614.13-4.13-1.13-4.622-3.817-.028-.16-.173-.358-.317-.43-1.154-.557-2.38-.842-3.652-.647-.974.15-1.657.82-1.925 1.73-.29.99-.393 2.035-.596 3.163.937.13 1.698.246 2.464.343 2.117.266 4.117.923 6.004 1.903.477.247.793.22 1.236-.07 2.57-1.688 5.793-.64 7.064 2.144.71 1.554-.018 3.753-1.684 4.966-.234.17-.482.53-.47.79.183 4.018-1.96 6.693-5.1 8.72-2.52 1.624-5.337 2.396-8.33 2.69-4.79.47-9.257-.362-13.298-3.04-2.427-1.608-4.08-3.785-4.553-6.736-.005-.03-.027-.06-.02-.085.414-1.59-.698-2.406-1.566-3.474-2.135-2.623-.243-5.67 2.28-6.542 1.588-.548 3.085-.226 4.8.958zm9.87-.666c-3.93-.092-8.132 1.347-10.653 3.525-2.672 2.31-3.826 5.586-1.7 8.884 1.006 1.56 2.438 2.637 4.063 3.467 4.158 2.122 8.543 2.405 13.037 1.466 2.8-.586 5.31-1.78 7.322-3.892 2.145-2.247 2.564-5.235.855-7.812-.718-1.084-1.712-2.07-2.774-2.828-2.928-2.084-6.3-2.808-10.15-2.81zm11.932-8.06c-1.24-.002-2.286.978-2.308 2.158-.02 1.173 1.11 2.263 2.332 2.242 1.21-.02 2.227-1.032 2.225-2.222 0-1.208-1-2.175-2.25-2.177zm3.493 13.734c1.178-.854 1.205-2.18.543-3.16-.678-1.005-2.338-1.628-3.7-.763l3.157 3.922zm-27.184-3.944c-1.373-.867-3.174-.107-3.706.807-.7 1.2-.453 2.406.592 3.112l3.113-3.92z\" ></path><path d=\"M199.832 209.57c-2.12.08-4.074-.433-5.885-1.552-.61-.378-.837-.864-.48-1.446.36-.59.974-.48 1.47-.16 1.146.736 2.394 1.06 3.734 1.26 2.073.306 3.967-.18 5.834-.956.318-.132.62-.444.922-.438.41.008.982.126 1.184.41.287.405.168.943-.383 1.25-1.99 1.105-4.104 1.717-6.395 1.633zM192.294 200.998c.028-1.087 1.04-2.013 2.142-1.958 1.092.054 2.057 1.053 2.035 2.108-.022 1.114-.96 1.972-2.137 1.958-1.208-.014-2.07-.904-2.04-2.108zM207.75 201.132c-.012 1.12-.94 1.994-2.098 1.977-1.154-.018-2.11-.936-2.102-2.022.008-1.03 1.043-2.03 2.115-2.05 1.198-.017 2.1.887 2.085 2.094z\" ></path></g><g fillRule=\"evenodd\" clipRule=\"evenodd\" ><path d=\"M189.897 195.724c2.914-1.658 6.07-2.337 9.31-2.542.12-.954.15-1.837.35-2.68.79-3.31 3.165-4.682 6.668-3.772 1.127.292 1.867.438 2.934-.515 2.31-2.062 5.874-.683 6.583 2.326.498 2.12-1.342 4.51-3.56 4.617-2.614.13-4.13-1.13-4.622-3.817-.028-.16-.173-.358-.317-.43-1.154-.557-2.38-.842-3.652-.647-.974.15-1.657.82-1.925 1.73-.29.99-.393 2.035-.596 3.163.937.13 1.698.246 2.464.343 2.117.266 4.117.923 6.004 1.903.477.247.793.22 1.236-.07 2.57-1.688 5.793-.64 7.064 2.144.71 1.554-.018 3.753-1.684 4.966-.234.17-.482.53-.47.79.183 4.018-1.96 6.693-5.1 8.72-2.52 1.624-5.337 2.396-8.33 2.69-4.79.47-9.257-.362-13.298-3.04-2.427-1.608-4.08-3.785-4.553-6.736-.005-.03-.027-.06-.02-.085.414-1.59-.698-2.406-1.566-3.474-2.135-2.623-.243-5.67 2.28-6.542 1.588-.548 3.085-.226 4.8.958zm9.87-.666c-3.93-.092-8.132 1.347-10.653 3.525-2.672 2.31-3.826 5.586-1.7 8.884 1.006 1.56 2.438 2.637 4.063 3.467 4.158 2.122 8.543 2.405 13.037 1.466 2.8-.586 5.31-1.78 7.322-3.892 2.145-2.247 2.564-5.235.855-7.812-.718-1.084-1.712-2.07-2.774-2.828-2.928-2.084-6.3-2.808-10.15-2.81zm11.932-8.06c-1.24-.002-2.286.978-2.308 2.158-.02 1.173 1.11 2.263 2.332 2.242 1.21-.02 2.227-1.032 2.225-2.222 0-1.208-1-2.175-2.25-2.177zm3.493 13.734c1.178-.854 1.205-2.18.543-3.16-.678-1.005-2.338-1.628-3.7-.763l3.157 3.922zm-27.184-3.944c-1.373-.867-3.174-.107-3.706.807-.7 1.2-.453 2.406.592 3.112l3.113-3.92z\" ></path><path d=\"M199.832 210.57c-2.12.08-4.074-.433-5.885-1.552-.61-.378-.837-.864-.48-1.446.36-.59.974-.48 1.47-.16 1.146.736 2.394 1.06 3.734 1.26 2.073.306 3.967-.18 5.834-.956.318-.132.62-.444.922-.438.41.008.982.126 1.184.41.287.405.168.943-.383 1.25-1.99 1.105-4.104 1.717-6.395 1.633z\" ></path><path fill=\"#F00\" d=\"M192.294 201.998c.028-1.087 1.04-2.013 2.142-1.958 1.092.054 2.057 1.053 2.035 2.108-.022 1.114-.96 1.972-2.137 1.958-1.208-.014-2.07-.904-2.04-2.108zM207.75 202.132c-.012 1.12-.94 1.994-2.098 1.977-1.154-.018-2.11-.936-2.102-2.022.008-1.03 1.043-2.03 2.115-2.05 1.198-.017 2.1.887 2.085 2.094z\" ></path></g>"`;
